### ISSUE TITLE
Crash for recursive call on `setCollectionViewLayout(_:animated:)`

### ISSUE THREAD

**Issue by dan12411** [kishikawakatsumi/IBPCollectionViewCompositionalLayout#129]
I have a feature to switch layout ( grid <-> list), so I call setCollectionViewLayout to switch. 
But if I start from 'grid -> list -> grid', it will crash.
From the call stack, we can see two IBPUICollectionViewCompositionalLayout is called repeatedly.

p.s. in iOS version below iOS 13.

Example code as follow,
```
    switch layoutMode {
    case .grid:
      collectionView.setCollectionViewLayout(grid, animated: false)
    case .list:
      collectionView.setCollectionViewLayout(list, animated: false)
    }
```

call stack & error message as follow,
<img width="1278" alt="截圖 2019-12-03 下午3 29 34" src="https://user-images.githubusercontent.com/21169170/70029342-d4c92b80-15e1-11ea-82fe-02c8220c363a.png">

**Comment by ali0201** [comment]
Hello, I also encountered this problem.

Have you solved it？

**Comment by ali0201** [comment]
> I have a feature to switch layout ( grid <-> list), so I call setCollectionViewLayout to switch.
> But if I start from 'grid -> list -> grid', it will crash.
> From the call stack, we can see two IBPUICollectionViewCompositionalLayout is called repeatedly.
> 
> p.s. in iOS version below iOS 13.
> 
> Example code as follow,
> 
> ```
>     switch layoutMode {
>     case .grid:
>       collectionView.setCollectionViewLayout(grid, animated: false)
>     case .list:
>       collectionView.setCollectionViewLayout(list, animated: false)
>     }
> ```
> 
> call stack & error message as follow,
> <img alt="截圖 2019-12-03 下午3 29 34" width="1278" src="https://user-images.githubusercontent.com/21169170/70029342-d4c92b80-15e1-11ea-82fe-02c8220c363a.png">

Hello, I also encountered this problem.

Have you solved it？

**Comment by ali0201** [comment]
I found the solution.
You can try like this.

```
- (void)setViewModel:(GGViewModel *)viewModel {
    self.collectionView.dataSource = nil;
    self.collectionView.delegate = nil;

    if (!self.collectionView.dataSource) {
        self.collectionView.dataSource = self;
    }
    if (!self.collectionView.delegate) {
        self.collectionView.delegate = self;
    }
    [self.collectionView setCollectionViewLayout:layout animated:NO];
}
```

**Comment by dan12411** [comment]
@ali0201 not yet! I will give it a try, thank you!

**Comment by dan12411** [comment]
@ali0201 
I tried to reset delegate for collectionView, it works! Thank you : )

```
    if #available(iOS 13.0, *) { } else {
      collectionView.delegate = nil
      if collectionView.delegate == nil {
        collectionView.delegate = self
      }
    }
    collectionView.setCollectionViewLayout(layout, animated: false)
```
